From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 23 Jan 2016 18:41:28 +0000
Subject: rfkill: Fix ABI change in 2.6.32.70
Forwarded: not-needed

struct rfkill changed, but as it's not defined in any header none of
the API users can depend on its definition.  Hide the changes from
genksyms.
---
 net/rfkill/core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/rfkill/core.c b/net/rfkill/core.c
index 85334a0..c42b2f4 100644
--- a/net/rfkill/core.c
+++ b/net/rfkill/core.c
@@ -49,6 +49,10 @@
 struct rfkill {
 	spinlock_t		lock;
 
+	/* bwh: Hide all changes from genksyms, since this type is opaque */
+#ifdef __GENKSYMS__
+	const char		*name;
+#endif
 	enum rfkill_type	type;
 
 	unsigned long		state;
@@ -72,7 +76,9 @@ struct rfkill {
 	struct delayed_work	poll_work;
 	struct work_struct	uevent_work;
 	struct work_struct	sync_work;
+#ifndef __GENKSYMS__
 	char			name[];
+#endif
 };
 #define to_rfkill(d)	container_of(d, struct rfkill, dev)
 
